src/esw/cameras.py

```diff
@@ -1,387 +1,232 @@
-            (camera_cmd.device, camera_cmd.resolution, stream) for stream, camera_cmd in enumerate(camera_commands)
+
+        primary_ip = rospy.get_param("cameras/ips/primary")
+        secondary_ip = rospy.get_param("cameras/ips/secondary")
+
+        self.ips = [primary_ip, secondary_ip]
+        self.cap_args = [
+            # [bps, width, height, fps]
+            list(rospy.get_param("cameras/arguments/worst_res")),
+            list(rospy.get_param("cameras/arguments/low_res")),
+            list(rospy.get_param("cameras/arguments/medium_res")),
+            list(rospy.get_param("cameras/arguments/high_res")),
+            list(rospy.get_param("cameras/arguments/best_res")),
         ]
+        self.skip_every_other_device = rospy.get_param("cameras/skip_every_other_device")
+
+    def get_num_devices_streaming(self) -> int:
+        num_streaming_devices = 0
+        for i in range(StreamManager.NUM_DIFFERENT_IPS):
+            for device in self.streamed_devices_by_port_by_laptop_idx[i]:
+                if device.device != -1:
+                    num_streaming_devices += 1
+        return num_streaming_devices
+
+    def get_available_stream(self, primary: bool) -> int:
+        idx = 0 if primary else 1
+        available_stream = 0
+        for i in range(StreamManager.MAX_NUM_POSSIBLE_CAMERAS):
+            if self.streamed_devices_by_port_by_laptop_idx[idx][available_stream].device == -1:
+                return available_stream
+            else:
+                available_stream += 1
+        rospy.logerr("CODE SHOULD NEVER REACH HERE")
+        return -1
+
+    def stop_device_id_stream(self, laptop_idx: int, device_id: int) -> None:
+        if self.stream_process_list[laptop_idx][device_id] is not None:
+            self.stream_process_list[laptop_idx][device_id].kill()
+            self.stream_process_list[laptop_idx][device_id].join()
+            self.stream_process_list[laptop_idx][device_id] = None
+        for i in range(StreamManager.NUM_DIFFERENT_IPS):
+            for j in range(len(self.streamed_devices_by_port_by_laptop_idx[i])):
+                if self.streamed_devices_by_port_by_laptop_idx[i][j].device == device_id:
+                    self.streamed_devices_by_port_by_laptop_idx[i][j] = CameraCmd(-1, -1)
+
+    def reset_cameras(self, req: ResetCamerasRequest) -> ResetCamerasResponse:
+        laptop_idx = 0 if req.primary else 1
+        for device in self.streamed_devices_by_port_by_laptop_idx[laptop_idx]:
+            if device.device != -1:
+                self.stop_device_id_stream(laptop_idx, device.device)
+
+        return ResetCamerasResponse(
+            self.streamed_devices_by_port_by_laptop_idx[0], self.streamed_devices_by_port_by_laptop_idx[1]
+        )
+
+    def handle_req(self, req: ChangeCamerasRequest) -> ChangeCamerasResponse:
+        cmds = req.camera_cmds
+        laptop_idx = 0 if req.primary else 1
+        device_id = cmds.device
+        cap = cmds.resolution
```

Comment:
Why is this variable called cap

Reply:
that's sort of a justin question, it should probably be renamed